### PR TITLE
Store token as Secret instead of String

### PR DESCRIPTION
**Pull request by bitwiseman**
# Description
Follow on to #326. We shouldn't keep the app installation token as a bare `String`.  
I don't think this is a huge deal given how `Secret` is actually stored and serialized, but this is more correct regardless.

# Submitter checklist
- [ ] Link to JIRA ticket in description, if appropriate.
- [ ] Change is code complete and matches issue description
- [ ] Automated tests have been added to exercise the changes
- [ ] Reviewer's manual test instructions provided in PR description. See Reviewer's first task below.

# Reviewer checklist
- [ ] Run the changes and verify that the change matches the issue description
- [ ] Reviewed the code
- [ ] Verified that the appropriate tests have been written or valid explanation given

# Documentation changes
- [ ] Link to jenkins.io PR, or an explanation for why no doc changes are needed

# Users/aliases to notify

### CODE REVIEW

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java [hunk 1]
```diff
@@ -115,6 +115,7 @@ public void setOwner(String owner) {
 
     @SuppressWarnings("deprecation") // preview features are required for GitHub app integration, GitHub api adds deprecated to all preview methods
     static AppInstallationToken generateAppInstallationToken(String appId, String appPrivateKey, String apiUrl, String owner) {
+        JenkinsJVM.checkJenkinsJVM();
```
Comment (bitwiseman) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java:118]: This method now only runs on a controller, let's enforce that.

File: src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java [hunk 2]
```diff
@@ -149,7 +150,8 @@ static AppInstallationToken generateAppInstallationToken(String appId, String ap
                 .create();
 
             long expiration = getExpirationSeconds(appInstallationToken);
-            AppInstallationToken token = new AppInstallationToken(appInstallationToken.getToken(),
+            AppInstallationToken token = new AppInstallationToken(
+                Secret.fromString(appInstallationToken.getToken()),
```
Comment (bitwiseman) [inline src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubAppCredentials.java:154]: The app installation token only exists as a bare string inside this method.

**Review by dwnusbaum — APPROVED** [review]

**Comment by dwnusbaum** [comment]
> I don't think this is a huge deal given how Secret is actually stored and serialized

Yeah, I don't think this affects the security of the implementation, but if it makes things simpler/less error prone then it seems fine to me.

**Review by jglick — APPROVED** [review]
Sure.

**Pull request merged**